### PR TITLE
feat(module.api): `getAccountHistory` api

### LINKED ISSUE #718: Get Account History: `/address/${address}/history/${height}/${txn-order}`
#### What would you like to be added:

Currently, we have List AccountHistory via `/address/${address}/history`. However, we need an endpoint to get single account history.

Endpoint: `/address/${address}/history/${height}/${txn-order}`
Return: `AddressHistory`

/triage accepted
/priority important-soon
/area module-api

### PR DESCRIPTION
**Pull request by canonbrother**
#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes #718

#### Additional comments?:
Current `getAccountHistory` doesn't support to get wallet kind transactions, eg: `sent`, `receive`, `blockReward`, if intended, will return undefined

### CODE REVIEW

File: packages/whale-api-client/__tests__/api/account.history.test.ts [hunk 1]
```diff
@@ -103,66 +103,104 @@ afterAll(async () => {
...
+      expect(acc?.owner).toStrictEqual(h.owner)
+      expect(acc?.txid).toStrictEqual(h.txid)
+      expect(acc?.txn).toStrictEqual(h.txn)
+    }
+  })
+
+  it('should get undefined as getting unsupport tx type - sent, received, blockReward', async () => {
+    const history = await client.address.listAccountHistory(colAddr, 30)
+    for (const h of history) {
+      if (['sent', 'receive', 'blockReward'].includes(h.type)) {
```
Comment (ivan-zynesis) [inline packages/whale-api-client/__tests__/api/account.history.test.ts:200]: Is the test actually setup with rewards in history?

```
const rewards = history.filter(h => h.type === 'blockReward')
expect(rewards.length).toBeGreaterThan(0)
```

**Review by ivan-zynesis — COMMENTED** [review]
It is actually quite annoying that upstream `listaccounthistory` return `sent` and `receive` (utxo transaction) too, unfortunately doesn't seems will be fixed anytime soon.

File: packages/whale-api-client/__tests__/api/account.history.test.ts (hunk 1 above)
Comment (canonbrother) [reply packages/whale-api-client/__tests__/api/account.history.test.ts]: checked `RegTestFoundationKeys[0].operator.address`'s `blockReward` history

**Review by ivan-zynesis — APPROVED** [review]

File: src/module.api/address.controller.e2e.ts [hunk 2]
```diff
@@ -209,6 +210,146 @@ describe('listAccountHistory', () => {
...
+      }
+      const acc = await controller.getAccountHistory(poolAddr, h.block.height, h.txn)
+      expect(acc?.owner).toStrictEqual(h.owner)
+      expect(acc?.txid).toStrictEqual(h.txid)
+      expect(acc?.txn).toStrictEqual(h.txn)
+    }
+  })
+
+  it('should get undefined as invalid height', async () => {
+    const acc = await controller.getAccountHistory(await container.getNewAddress(), Number(`${'0'.repeat(64)}`), 1)
```
Comment (ivan-zynesis) [inline src/module.api/address.controller.e2e.ts:330]: The number input here should be still zero, can test with NaN.

Nestjs doesn't seems to built in with positive number only pipe, maybe can test with negative number too.

**Review by ivan-zynesis — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**